src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static remove() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name, hidden = false) {
+    super(health, name);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(arg) {
+    if (arg instanceof Herbivore && !arg.hidden) {
```

Comment:
[CODE PERFORMANCE] It is preferred to filter out animals with health less than or equal to 0 from the `Animal.alive` array using Array's `filter` method. Instead of checking the health of the animal and then calling `Animal.remove()` in the `bite` method, you could directly use the `filter` method after adjusting the health of the animal to remove any that have died.